File: src/main/java/src/com/study/design/pattern/creational/builder/CodeBuilder.java
package src.com.study.design.pattern.creational.builder;

class CodeBuilder {

    private StringBuilder stringBuilder = new StringBuilder();

    CodeBuilder(String className) {
        stringBuilder.append("public class ");
        stringBuilder.append(className);
        stringBuilder.append(System.getProperty("line.separator"));
        stringBuilder.append("{");
        stringBuilder.append(System.getProperty("line.separator"));
    }

    CodeBuilder addField(String name, String type)
    {
        stringBuilder.append("  public ");
        stringBuilder.append(type);
        stringBuilder.append(" ");
        stringBuilder.append(name );
        stringBuilder.append(";");
        stringBuilder.append(System.getProperty("line.separator"));

        return this;
    }

    @Override
    public String toString()
    {
        stringBuilder.append("}");
        return stringBuilder.toString();
    }
}

File: src/main/kotlin/src/com/study/design/pattern/creational/builder/CodeBuilder.kt
package src.com.study.design.pattern.creational.builder

class CodeBuilder(private val className: String) {

    private val properties = mutableMapOf<String, String>()

    fun addField(name: String, type: String): CodeBuilder {
        properties[name] = type
        return this
    }

    override fun toString() =
        buildString {
            this.appendln("class $className (")

            properties.forEach {
                this.appendln("   val ${it.key}: ${it.value},")
            }

            this.append(")")

        }.reversed().replaceFirst(Regex(","), "").reversed()
}